pyproject.toml

```diff
@@ -62,16 +62,16 @@ dependencies = [
     "sphinxcontrib-htmlhelp>=2.0.0",
     "sphinxcontrib-serializinghtml>=1.1.9",
     "sphinxcontrib-qthelp",
-    "Jinja2>=3.0",
-    "Pygments>=2.14",
+    "Jinja2>=3.1",
+    "Pygments>=2.17",
     "docutils>=0.20,<0.22",
-    "snowballstemmer>=2.0",
-    "babel>=2.9",
+    "snowballstemmer>=2.2",
+    "babel>=2.13",
     "alabaster~=0.7.14",
     "imagesize>=1.3",
-    "requests>=2.25.0",
-    "packaging>=21.0",
-    "importlib-metadata>=4.8; python_version < '3.10'",
+    "requests>=2.30.0",
```

Comment:
This reminds me: this is a step along the path to removing a `linkcheck` builder `socket.setdefaulttimeout` call: https://github.com/sphinx-doc/sphinx/pull/11876#issuecomment-1889563435

(but also requires `urllib3` v2.x or greater, and I don't think we should require that yet)